fission-web-server/library/Fission/Web/Server/Types.hs

```diff
@@ -140,6 +155,102 @@ instance MonadDB (Transaction Server) Server where
 instance MonadRelayStore Server where
   getStoreVar = asks linkRelayStoreVar
 
+instance MonadPowerDNS Server where
+  getClientEnv = do
+    apiUrl <- asks pdnsURL
+    apiKey <- asks pdnsApiKey
+    uri <- parseBaseUrl $ Text.unpack (textDisplay apiUrl)
+    let
+      rawHTTPSettings =
+        case baseUrlScheme uri of
+          Http  -> defaultManagerSettings
+          Https -> tlsManagerSettings
+
+    mgr <- liftIO $ newManager rawHTTPSettings
+    return $ PDNS.applyXApiKey (textDisplay apiKey) (mkClientEnv mgr uri)
+
+  set recType url zoneTxt contents ttl = do
+    logDebug $ mconcat
+      [ "Set DNS "
+      , displayShow recType
+      , " Record "
+      , displayShow url
+      , " to "
+      , displayShow contents
+      ]
+    clientEnv <- getClientEnv
+    let
+      rrset = RRSets [RRSet { rrset_name = Text.append (textDisplay url) "."
+                      , rrset_type = recType
+                      , rrset_ttl = ttl
+                      , rrset_changetype = Just Replace
+                      , rrset_records = Just $ formatRecords contents
+                      , rrset_comments = Nothing
+      }]
+    resp <- liftIO $ runClientM (PDNS.updateRecords "localhost" zoneTxt rrset) clientEnv
```

Comment:
Hardcoded `localhost`?

Reply:
yeah, this is weird, technically you can use the pdns http API to control multiple pdns servers ... we're likely only going to ever control the local one. (i.e. this doesn't mean run anything against the server's localhost it's actually part of the url path in all REST calls).

Reply:
WUT